### PR TITLE
[paper] clean up numbers section

### PR DESCRIPTION
**Pull request by enricozb**

### CODE REVIEW

File: paper/PAPER.typst [hunk 1]
```diff
@@ -41,20 +41,22 @@ HVM2's syntax has seven different types of _agents_ (in Lafont's terms) or _Node
   ```
   <Node> ::=
       | "*"                    -- (ERA)ser
-      | "@" <name>             -- (REF)erence
-      | <number>               -- (NUM)ber
+      | "@" <alphanumeric>     -- (REF)erence
```
Comment (developedby) [inline paper/PAPER.typst:44]: If you want to be more specific about "alphanumeric", the regex we're using is `[a-zA-Z0-9_.-/]+`

Comment (enricozb) [reply paper/PAPER.typst:44]: Can defs start with digits? `@0 = ..`?

Comment (developedby) [reply paper/PAPER.typst:44]: yup

**Review by VictorTaelin — APPROVED** [review]
Looks fine overall. Note that some numeric ops names differ (FP_SUB vs FLIP_SUB).

**Pull request merged**